packages/mui-codemod/src/deprecations/button-classes/test-cases/expected.js

```diff
@@ -0,0 +1,673 @@
+          color: 'red',
+        },
+
+        [`&.${buttonClasses.text}.${buttonClasses.colorWarning}`]: {
+          color: 'red',
+        },
+
+        [`&.${buttonClasses.outlined}.${buttonClasses.colorInherit}`]: {
+          color: 'red',
+        },
+
+        [`&.${buttonClasses.outlined}.${buttonClasses.colorPrimary}`]: {
+          color: 'red',
+        },
+
+        [`&.${buttonClasses.outlined}.${buttonClasses.colorSecondary}`]: {
+          color: 'red',
+        },
+
+        [`&.${buttonClasses.outlined}.${buttonClasses.colorSuccess}`]: {
+          color: 'red',
+        },
+
+        [`&.${buttonClasses.outlined}.${buttonClasses.colorError}`]: {
+          color: 'red',
+        },
+
+        [`&.${buttonClasses.outlined}.${buttonClasses.colorInfo}`]: {
+          color: 'red',
+        },
+
+        [`&.${buttonClasses.outlined}.${buttonClasses.colorWarning}`]: {
+          color: 'red',
+        },
+
+        [`&.${buttonClasses.contained}.${buttonClasses.colorInherit}`]: {
+          color: 'red',
+        },
+
+        [`&.${buttonClasses.contained}.${buttonClasses.colorPrimary}`]: {
+          color: 'red',
+        },
+
+        [`&.${buttonClasses.contained}.${buttonClasses.colorSecondary}`]: {
+          color: 'red',
+        },
+
+        [`&.${buttonClasses.contained}.${buttonClasses.colorSuccess}`]: {
+          color: 'red',
+        },
+
+        [`&.${buttonClasses.contained}.${buttonClasses.colorError}`]: {
+          color: 'red',
+        },
+
+        [`&.${buttonClasses.contained}.${buttonClasses.colorInfo}`]: {
+          color: 'red',
+        },
+
+        [`&.${buttonClasses.contained}.${buttonClasses.colorWarning}`]: {
+          color: 'red',
+        },
+        [`&.${buttonClasses.text}.${buttonClasses.sizeSmall}`]: {
+          color: 'red',
+        },
+        [`&.${buttonClasses.text}.${buttonClasses.sizeMedium}`]: {
+          color: 'red',
+        },
+        [`&.${buttonClasses.text}.${buttonClasses.sizeLarge}`]: {
+          color: 'red',
+        },
+        [`&.${buttonClasses.outlined}.${buttonClasses.sizeSmall}`]: {
+          color: 'red',
+        },
+        [`&.${buttonClasses.outlined}.${buttonClasses.sizeMedium}`]: {
+          color: 'red',
+        },
+        [`&.${buttonClasses.outlined}.${buttonClasses.sizeLarge}`]: {
+          color: 'red',
+        },
+        [`&.${buttonClasses.contained}.${buttonClasses.sizeSmall}`]: {
+          color: 'red',
+        },
+        [`&.${buttonClasses.contained}.${buttonClasses.sizeMedium}`]: {
+          color: 'red',
+        },
+        [`&.${buttonClasses.contained}.${buttonClasses.sizeLarge}`]: {
+          color: 'red',
+        },
+        [`&.${buttonClasses.sizeSmall}${" > "}.${buttonClasses.icon}`]: {
+          color: 'red',
+        },
+        [`&.${buttonClasses.sizeMedium}${" > "}.${buttonClasses.icon}`]: {
+          color: 'red',
+        },
+        [`&.${buttonClasses.sizeLarge}${" > "}.${buttonClasses.icon}`]: {
```

Comment:
tried transforming to `&.${buttonClasses.sizeLarge} > .${buttonClasses.icon}` but couldn't succeed. Does this tranformtion works or shall try again?

Reply:
It should work if you add `' > '` to the `quasis` instead of the `expressions` of the template. The `quasis` are the chunks of the template string that are not within `${}`.

Reply:
Done, PR is ready for review